api/apps/geoprocessing/src/marxan-sandboxed-runner/progress.ts

```diff
@@ -0,0 +1,18 @@
+export class Progress {
+  private previousChunk = '';
+  private outOf?: number;
+  private runsMade: Record<number, true> = {};
+
+  read(chunk: Buffer): number {
+    const current = chunk.toString();
+    const joined = this.previousChunk + current;
+    const matches = joined.matchAll(/Run (\d+) is finished \(out of (\d+)\)./g);
+    this.previousChunk = current;
+    for (const match of matches) {
+      const [, currentRun, currentTotal] = match;
+      this.outOf ??= +currentTotal;
```

Comment:
I believe it should rather base on pre-defined number of desired runs (derived form `input.dat`)

Reply:
for simplicity's sake we can even keep this as is - in practice (unless there's a bug in the Marxan CLI) the regex match should be equivalent to getting data from `input.dat`, but I like the idea of this being self-contained as is now.